### DISCUSSION TITLE
Documenting extra status codes

### DISCUSSION THREAD

**Discussion by sondrelg** [awtkns/fastapi-crudrouter#102] [General]
Hi!

I was just testing out [schemathesis](https://github.com/schemathesis/schemathesis) on a new project implementing crudrouter, and found it complaining about the fact that one of my generated endpoints had undocumented response codes.

Output:

```
DELETE /api/v1/authors/{item_id} [P]
1. Received a response with a status code, which is not defined in the schema: 404

Declared status codes: 200, 422
```

This error is _technically_ correct - there is no 404 documented, but I can definitely understand how crudrouter as a library might _not_ want to auto-document every possible status code, since it would make the schema way too detailed/verbose.

I was just wondering, is it _possible_ to opt-into documenting extra status codes? Or is it possible to manually extend the status codes documented? 

If this feature doesn't exist yet, do you think it would be something worth implementing? If nothing else it would be cool to be able to switch it on just for testing.

**Comment by awtkns** [comment]
Hi @sondrelg, thanks for the detailed write up! I do think it would be documenting theses extra response codes. As of now, there are only three codes returned: `200, 404 and 422`.

Edit: I've created issue #103 to track this, is this something you'd be interested in PRing?

**Reply by sondrelg** [reply]
So would it just be a matter of making sure 404s are documented? 🙂  I wouldn't mind giving it a try at least, if so 👍

**Reply by awtkns** [reply]
> So would it just be a matter of making sure 404s are documented? 🙂 I wouldn't mind giving it a try at least, if so 👍

Awesome let me know if you need help :).  Schemathesis should pass after this is merged.

**Comment by antonslam** [comment]


**Reply by awtkns** [reply]
Hi @antonslam would you be able to transfer this to a separate discussion as this seems unrelated to the first post. Also, I would be better able to answer your question if you could provide a minimal runnable example that reproduces the problem. What is the error that is shown in logs / status code when the faulty put request is made?
